### ISSUE TITLE
Implement getters and setters for base slots in HDF5

### ISSUE THREAD

**Issue by rcannood** [scverse/anndataR#58]
Implement all getters and setters in HDF5 for X, obs, var, obs_names, var_names, layers.

Encompasses:

- #16
- #14
- #15
- #19

**Comment by rcannood** [comment]
@lazappi would you like some help with this?

**Comment by lazappi** [comment]
I'm planning on working on this today. What was holding me up was #52 and #51. The reading is mostly done but I think these need to be sorted before too much work can be done on the writing side.

**Comment by rcannood** [comment]
Fantastic! Please review #65, #67, #68 as this may be of help to you. Having said that, feel free to scratch anything included in #68 as sometimes it's more work to adapt something that's broken than to start from scratch ;)

In discussing with @LouiseDck yesterday, I'm thinking that the only way to properly make sure that anndataR is able to read files created by anndata and vice versa, is to use reticulate to implement tests like:

```r
test_that("test Python -> h5ad -> R", {
  # create anndata in python
  ad <- anndata::AnnData(X = X, obs = obs, var = var)
  
  # write to file
  ad$write_h5ad("file.h5ad")

  # read from file
  ad_new <- HDF5AnnData$new("file.h5ad")
  
  # check whether ad_new$obs == obs and so on
})


test_that("test R -> h5ad -> Python", {
  # create anndata in R
  ad <- InMemoryAnnData$new(X = X, obs = obs, var = var)
  
  # write to file
  ad$write_h5ad("file.h5ad")

  # read from file, hopefully Python anndata is able to read this
  ad_new <- anndata::read_h5ad("file.h5ad")
  
  # check whether ad_new$obs == obs and so on
})
```

Even without the extra checks, seeing if the python anndata is able to read files produced by anndataR is already very valuable.

**Comment by rcannood** [comment]
Completed with #69!
